python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -646,41 +677,43 @@ cdef class Column:
         strides, size_type overflow) by a prior call to
         `_prepare_array_metadata`.
         """
-        if len(shape) == 1:
-            size = shape[0]
-            return Column(dtype, size, data, None, 0, 0, [])
-
-        else:
-            num_rows, num_cols = shape
+        ndim = len(shape)
+        flat_size = functools.reduce(operator.mul, shape)
+        data_col = Column(
```

Comment:
```suggestion
        current_data_col = Column(
```
Nit: I think if you use this assignment, you don't need the `current_data_col = data_col` below`